docs/sections/workflows/writing_basics.md

```diff
@@ -175,103 +154,445 @@ See the {ref}`corresponding topics section for more details <topics:processes:us
+Let get started by creating a file for our work chain (e.g. `multiply_add_workchain.py`), and adding the following piece of code:
 
-You can give the work chain any valid Python class name, but the convention is to have it end in {class}`~aiida.engine.processes.workchains.workchain.WorkChain` so that it is always immediately clear what it references.
-Let's go over the methods of the `MultiplyAddWorkChain` one by one:
+```{code-block} python
+from aiida.orm import Int
+from aiida.engine import WorkChain
 
-(workflows-writing-basics-define)=
+class MultiplyAddWorkChain(WorkChain):
+    """WorkChain to multiply two numbers and add a third, for testing and demonstration purposes."""
 
-```{literalinclude} include/code/multiply_add.py
-:language: python
-:pyobject: MultiplyAddWorkChain.define
-:dedent: 4
+    @classmethod
+    def define(cls, spec):
+        """Specify inputs and outputs."""
+        super().define(spec)
 
+        spec.input("x", valid_type=Int)
+        spec.input("y", valid_type=Int)
+        spec.input("z", valid_type=Int)
+
+        ...
 ```
 
+Writing a work chain in AiiDA requires creating a class that inherits from the {class}`~aiida.engine.processes.workchains.workchain.WorkChain` class, as shown above.
+You can give the work chain any valid Python class name, but the convention is to have it end in {class}`~aiida.engine.processes.workchains.workchain.WorkChain` so that it is always immediately clear what it references.
+Hence the chosen name `MultiplyAddWorkChain`.
+
 The most important method to implement for every work chain is the `define()` method.
-This class method must always start by calling the `define()` method of its parent class.
-Next, the `define()` method should be used to define the specifications of the work chain, which are contained in the work chain `spec`:
+This class method must always start by calling the `define()` method of its parent class (the `super()` function referes to the parent class).
+The `define()` method should be used to define the specifications of the work chain, which are contained in the work chain `spec`:
 
 * the **inputs**, specified using the `spec.input()` method.
   The first argument of the `input()` method is a string that specifies the label of the input, e.g. `'x'`.
   The `valid_type` keyword argument allows you to specify the required node type of the input.
   Other keyword arguments allow the developer to set a default for the input, or indicate that an input should not be stored in the database, see {ref}`the process topics section <topics:processes:usage:spec>` for more details.
-* the **outline** or logic of the workflow, specified using the `spec.outline()` method.
+
+:::{note}
+
+All inputs and outputs of a work chain must be of AiiDA data types because only then they can be stored as Nodes in the AiiDA database.
+
+:::
+
+Next, let us add an outline to our work chain, i.e., a sequence of steps and the logic of the work chain (please modify your script along):
```

Comment:
In my opinion, they should put their hands on it. Even better than copying it would be if they were to write it down following the instruction and example. I know you are worried about possible problems during the tutorial, but I think they are part of the process.